core/paddlefl_mpc/mpc_protocol/abstract_context.h

```diff
@@ -0,0 +1,182 @@
+// Copyright (c) 2020 PaddlePaddle Authors. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+#pragma once
+
+#include <algorithm>
+#include <memory>
+
+#include "core/paddlefl_mpc/mpc_protocol/abstract_network.h"
+#include "core/privc3/prng_utils.h"
+#include "paddle/fluid/platform/enforce.h"
+
+namespace paddle {
+
+namespace mpc {
+
+using block = psi::block;
+using PseudorandomNumberGenerator = psi::PseudorandomNumberGenerator;
+
+class AbstractContext {
+public:
+  AbstractContext() = default;
+  AbstractContext(const AbstractContext &other) = delete;
+
+  AbstractContext &operator=(const AbstractContext &other) = delete;
+
+  virtual void init(size_t party, std::shared_ptr<AbstractNetwork> network, block seed,
```

Comment:
it seems no common method use the seed and seed2. so you can put the seed in derived classes' ctors, and keep the init as concise as init(party, network)